sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -114,7 +114,7 @@ private[sql] object DataFrame {
 @Experimental
 class DataFrame private[sql](
     @transient val sqlContext: SQLContext,
-    @DeveloperApi @transient val queryExecution: SQLContext#QueryExecution) extends Serializable {
+    @DeveloperApi @transient val queryExecution: QueryExecution) extends Serializable {
```

Comment:
Should we preserve binary compatibility for developer API? At least MIMA doesn't complain about it...